### PR TITLE
ALB as internal/internet facing 🥅 

### LINKED ISSUE #13: Allow for setting the ALB to internal or public
### Background
`dibbs-aws` currently creates a public facing load balancer.

### Action requested
Enable configuration option to set the ALB as internal or public

### Acceptance Criteria
- ALB should default to internal
- Only require passing a flag if ALB to be public

### Additional context
- Feature request for client implementation

### PR DESCRIPTION
**Pull request by alismx**
# DEVOPS PULL REQUEST

## Related Issue

- resolve #13 

## Changes Proposed

- Allow for internal or public ALB, defaults to internal
- update docs and formatting

## Additional Information

- This is a feature request for a client implemenation

### CODE REVIEW

**Review by shanice-skylight — COMMENTED** [review]
The code looks straight forward gtg on that. Is there an intended terraform output or simply adding the parameter? I ran terraform plan in the ecs directory and was prompted for a name of an App Mesh.

**Comment by alismx** [comment]
> The code looks straight forward gtg on that. Is there an intended terraform output or simply adding the variable? I ran terraform plan in the ecs directory and was prompted for a name of an App Mesh.

No output. 

If you'd like, we can pair tomorrow and we can go over the requirements to run the terraform. A tfvars file is needed.

**Review by shanice-skylight — APPROVED** [review]
LGTM

File: terraform/implementation/ecs/README.md [hunk 1]
```diff
@@ -31,6 +31,7 @@
 
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
+| <a name="input_alb_internal"></a> [alb\_internal](#input\_alb\_internal) | Whether the ALB is public or private | `bool` | `true` | no |
```
Comment (rin-skylight) [inline terraform/implementation/ecs/README.md:34]: Non-blocking, for your consideration: the concepts of "internal" and "private" may be disjoint to certain end users or jurisdictions. You may want to consider buffing the description to include something like, "Whether the ALB is _public_ (intended for external access) or _private_ (only intended to be accessed within your AWS private cloud)."

File: terraform/modules/ecs/README.md [hunk 2]
```diff
@@ -56,6 +56,7 @@ No modules.
 
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
+| <a name="input_alb_internal"></a> [alb\_internal](#input\_alb\_internal) | Flag to determine if the ALB is public | `bool` | `true` | no |
```
Comment (rin-skylight) [inline terraform/modules/ecs/README.md:59]: See above comment regarding the concepts of private vs internal.

File: terraform/modules/ecs/_variable.tf [hunk 3]
```diff
@@ -1,3 +1,8 @@
+variable "alb_internal" {
+  type        = bool
+  description = "Flag to determine if the ALB is public"
```
Comment (rin-skylight) [inline terraform/modules/ecs/_variable.tf:3]: See above comment regarding the concepts of private vs internal.

**Review by rin-skylight — APPROVED** [review]
This looks great! I left an optional consideration for your review, but that does not block this PR from going in.

File: terraform/implementation/ecs/README.md (hunk 1 above)
Comment (alismx) [reply terraform/implementation/ecs/README.md:34]: I'll include these clarifications on the next PR!

**Pull request merged**